2022q3/freebsd-foundation.adoc

```diff
@@ -0,0 +1,170 @@
+  event.  A concurrent callout_cc_add() call might be changing the next event
+  time, and the race can cause handleevents() to program an out-of-date time,
+  causing the callout to run later (by an unbounded period, up to the idle
+  hardclock period of 1s) than requested.
+
+===== Bhyve Issue Support
+
+The Foundation contracted John Baldwin to dedicate time to Bhyve as issues
+arise, especially security issues.  Here is a summary of his 2022q3 work on that
+contract.
+
+- bb31aee bhyve virtio-scsi: Avoid out of bounds accesses to guest requests.
+- 62806a7 bhyve virtio-scsi: Tidy warning and debug prints.
+- 7afe342 bhyve e1000: Sanitize transmit ring indices.
+- c94f30e bhyve: Validate host PAs used to map passthrough BARs.
+- 16bedf5 pci: Add helper routines to iterate over a device's BARs.
+- baf753c bhyve: Support other schemes for naming pass-through devices.
+- fa46f37 bhyve e1000: Skip packets with a small header.
+- e7439f6 bhyve xhci: Cache the value of MaxPStreams when initializing an endpoint.
+
+===== RISC-V Improvements
+
+At the end of the quarter, the Foundation contracted Mitchell Horne to add and
+improve support for RISC-V hardware.  Mitchell will also perform general
+maintenance such as fixing bugs, handling reports, providing review for new code
+changes, and improving source code legibility and documentation.
+
+==== Continuous Integration and Quality Assurance
+
+The Foundation provides a full-time staff member and funds projects to improve
+continuous integration, automated testing, and overall quality assurance efforts
+for the FreeBSD project.  You can read about CI activities this quarter in a
+dedicated entry.
+
+==== FreeBSD Advocacy and Education
+
+Much of our effort is dedicated to Project advocacy.  This may involve
+highlighting interesting FreeBSD work, producing literature and video tutorials,
+attending events, or giving presentations.  The goal of the literature we
+produce is to teach people FreeBSD basics and help make their path to adoption
+or contribution easier.  Other than attending and presenting at events, we
+encourage and help community members run their own FreeBSD events, give
+presentations, or staff FreeBSD tables.
+
+The FreeBSD Foundation sponsors many conferences, events, and summits around the
+globe.  These events can be BSD-related, open source, or technology events
+geared towards underrepresented groups.  We support the FreeBSD-focused events
+to help provide a venue for sharing knowledge, working together on projects, and
+facilitating collaboration between developers and commercial users.  This all
+helps provide a healthy ecosystem.  We support the non-FreeBSD events to promote
+and raise awareness of FreeBSD, to increase the use of FreeBSD in different
+applications, and to recruit more contributors to the Project.  We are
+continuing to attend virtual events and planning the June 2022 Developer Summit.
```

Comment:
Is "June 2022" correct? This is the July-to-September report.

Reply:
No.  That should say the November 2022 Developer Summit.  Will fix now.

Reply:
Never mind. It was fixed in the 2nd commit.

Reply:
Are you looking at what's in the file after the second commit?  It shouldn't still say June.